### DISCUSSION TITLE
Unexpected behaviour of "Sync annotations" feature

### DISCUSSION THREAD

**Discussion by DerMasri** [foxglove/community#902] [Q&A]
@
[sync_problem.zip](https://github.com/foxglove/community/files/13819527/sync_problem.zip)

Hi together =)

I have a question regarding the "Sync annotations" feature. I have generated a test mcap file.
The annotations & image have exactly the same timestamp. Also the timestamps of the mcap entry are identical.

Turning the feature on, gives me "Waiting for synchronized annotations…" in the image panel all the time.
What am I doing wrong? Or is the feature currently not working properly?

Thanks a lot =)

**Comment by snosenzo** [comment]
@DerMasri Thanks for filing and providing some data. I tried to reproduce your issue with the data you provided and wasn't able to. Indeed, when first enabling sync annotations it will show the "Waiting for synchronized annotations..." message (this is expected), but after starting playback and receiving another set of messages in the image panel it looks like it's working as expected and both topics show up. 

It might help further by first confirming you are on the latest version of studio (1.83.0) and afterwards providing a video of this issue happening and perhaps the layout you are using?

**Reply by DerMasri** [reply]
Wow it seems to be a very rare issue. I tested a bit around.
It happens for example when you load a new mcap file with different topic names. I have uploaded a test-set with two different mcap files:
[sync_problem.zip](https://github.com/foxglove/community/files/13822181/sync_problem.zip)


It contains also a video of how I got stuck:
- create a new image panel
- load sync_problem.mcap -> sync on is working
- load sync_problem2.mcap -> change topic name -> sync is not working any more
- To break out -> create a new image panel

I was not able to reproduce another constellation where I had the same issue.. (based on reloading a measurement with the same name, but different content).

Version: 1.83.0
(Kernel: 5.15.0-91-generic)

**Reply by DerMasri** [reply]
Random guess: without knowing the code, it seems that the internal states of the sync-mechanism need to be cleaned as soon new data is loaded.

**Reply by jtbandes** [reply]
Thanks for providing the reproduction steps. I've converted this to an issue here: https://github.com/foxglove/community/issues/908
